### ISSUE TITLE
Fix S6580 FP/FN: Do not raise when DateTimeStyles.RoundtripKind is specified

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#8004]
### Description

When using one of the following methods with the overload that accepts a [DateTimeStyles](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.datetimestyles?view=net-7.0#system-globalization-datetimestyles-roundtripkind) argument and `DateTimeStyles.RoundtripKind` is used, a format provider does not need to be specified.

### Repro steps

```csharp

using System.Globalization;

var before = new DateTimeOffset(2008, 4, 12, 9, 30, 0, new TimeSpan(-8, 0, 0));
// converted to string with the "o" ISO round-trip-style. DateTimeKind is "baked" in the string.
var stringified = before.ToString("o"); 
// RoundTripKind is enough to parse this.
var after = DateTimeOffset.Parse(stringified, null, DateTimeStyles.RoundtripKind); // Noncompliant FP
Console.WriteLine("Round-tripped {0} to {1}.", before, after); // before == after
```

You can actually see this on Microsoft's msdn code snippet about the "o", ISO-roundtrip style format ([click me](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#Roundtrip))

Applies to overloads of these methods:

[DateTime.Parse](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.parse?view=net-7.0)
[DateTimeOffset.Parse](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.parse?view=net-7.0)
[DateTime.TryParse](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tryparse?view=net-7.0)
[DateTimeOffset.TryParse](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tryparse?view=net-7.0)
[DateTime.ParseExact](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.parseexact?view=net-7.0)
[DateTimeOffset.ParseExact](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.parseexact?view=net-7.0)
[DateTime.TryParseExact](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tryparseexact?view=net-7.0)
[DateTimeOffset.TryParseExact](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tryparseexact?view=net-7.0)

**Comment by BennieCopeland** [comment]
In addition to this, I don't think it should raise when I provide the expected format/s:

```csharp
var before = new DateOnly(2008, 4, 12);
var stringified = before.ToString("o");
// Unfortunately, RoundTripKind doesn't work with DateOnly
var dt = DateOnly.ParseExact(stringified, "yyyy-MM-dd")
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1689](https://sonarsource.atlassian.net/browse/NET-1689)

[NET-1689]: https://sonarsource.atlassian.net/browse/NET-1689?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
